src/clds_hash_table.c

```diff
@@ -929,6 +1029,247 @@ CLDS_HASH_TABLE_ITEM* clds_hash_table_find(CLDS_HASH_TABLE_HANDLE clds_hash_tabl
+                                            if (list_item_count == 0)
+                                            {
+                                                // skip
+                                            }
+                                            else
+                                            {
+                                                /* Codes_SRS_CLDS_HASH_TABLE_42_026: [ clds_hash_table_snapshot shall call clds_sorted_list_get_all with the next portion of the allocated array. ]*/
+                                                CLDS_SORTED_LIST_GET_ALL_RESULT get_all_result = clds_sorted_list_get_all(current_bucket_array->hash_table[i], clds_hazard_pointers_thread, list_item_count, items_to_return + result_index);
+                                                if (get_all_result != CLDS_SORTED_LIST_GET_ALL_OK)
+                                                {
+                                                    /* Codes_SRS_CLDS_HASH_TABLE_42_061: [ If there are any other failures then clds_hash_table_snapshot shall fail and return CLDS_HASH_TABLE_SNAPSHOT_ERROR. ]*/
+                                                    LogError("clds_sorted_list_get_all failed with %" PRI_MU_ENUM, MU_ENUM_VALUE(CLDS_SORTED_LIST_GET_ALL_RESULT, get_all_result));
+                                                    failed = true;
+                                                }
+                                                else
+                                                {
+                                                    result_index += list_item_count;
+                                                }
+                                            }
+                                        }
+                                    }
+
+                                    /* Codes_SRS_CLDS_HASH_TABLE_42_027: [ clds_hash_table_snapshot shall call clds_sorted_list_unlock_writes. ]*/
+                                    clds_sorted_list_unlock_writes(current_bucket_array->hash_table[i]);
+                                }
+                            }
+                        }
+
+                        current_bucket_array = next_bucket_array;
+                    }
+
+                    need_to_unlock_all = false; // Unlocked
+
+                    if (failed)
+                    {
+                        result = CLDS_HASH_TABLE_SNAPSHOT_ERROR;
+
+                        for (uint64_t i = 0; i < result_index; i++)
+                        {
+                            clds_sorted_list_node_release(items_to_return[i]);
+                        }
+                        free(items_to_return);
+                    }
+                    else
+                    {
+                        /* Codes_SRS_CLDS_HASH_TABLE_42_028: [ clds_hash_table_snapshot shall store the allocated array of items in items. ]*/
+                        *items = (CLDS_HASH_TABLE_ITEM**)items_to_return;
```

Comment:
>(CLDS_HASH_TABLE_ITEM**) [](start = 33, length = 24)

Likely not needed

Reply:
Casting from sorted list item is an incompatible type

---
In reply to: [352379595](https://github.com/dcristoloveanu/clds/pull/33#discussion_r352379595) [](ancestors = 352379595)